aten/src/TH/generic/THTensorMath.c

```diff
@@ -3712,6 +3726,36 @@ TENSOR_IMPLEMENT_LOGICAL_SUM(logicalany, ||, 0)
 #define TH_MATH_NAME(fn) fn
 #endif
 
+#ifdef _OPENMP
+LAB_IMPLEMENT_BASIC_FUNCTION(log,TH_MATH_NAME(log),HYPER_TH_OMP_OVERHEAD_THRESHOLD)
```

Comment:
Is there something we can do to avoid duplicating definitions for all of these pointwise operations?

Reply:
Maybe we have to modify the `LAB_IMPLEMENT_BASIC_FUNCTION` Macro when openmp is unavailable. How about adding a default OMP_OVERHEAD_THRESHOLD to the macro? It is a little annoying to pass a additional parameter when using the macro.

Reply:
An idiom which we have used in `interned_strings.h` looks something like this:

```
  #define FORALL_JIT_SYMBOLS(_) \
  _(Assign) \
  _(Constant) \
  _(CppOp) \
  _(Drop) \
  _(Eval) \
  _(Expand) \
```

and then

```
    enum BuiltinSymbol {
      #define DEFINE_SYMBOL(s) \
        k##s,
      FORALL_JIT_SYMBOLS(DEFINE_SYMBOL)
      #undef DEFINE_SYMBOL
      kLastSymbol, //where we start counting for new symbols
    };
```

It seems like you might to apply this principle here, where `LAB_IMPLEMENT_BASIC_FUNCTION` is the parameterized function, and it discards the threshold argument depending on the conditional.

Reply:
@ezyang [C macros with optional parameters](https://stackoverflow.com/questions/3046889/optional-parameters-with-c-macros) is a nice idea to handle with the situation more flexibly.